### PR TITLE
Support arm64 instances

### PR DESCRIPTION
**Pull request by stevehodgkiss**
Added an arm64 binary of ejson2env to be used if the architecture matches.

### CODE REVIEW

File: hooks/post-checkout [hunk 1]
```diff
@@ -3,7 +3,14 @@
 set -euo pipefail
 
 EJSON2ENV_PLUGIN_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")"; cd ..; pwd)
-EJSON2ENV_BIN_PATH="$EJSON2ENV_PLUGIN_ROOT/ejson2env"
+arch="${BUILDKITE_PLUGIN_EJSON2ENV_ARCH:-}"
```
Comment (orien) [inline hooks/post-checkout]: I'm not sure I understand the intent of this arch parameter. If a value is provided it seems to be overridden within the else block below.

File: hooks/post-checkout [hunk 2]
```diff
@@ -3,7 +3,14 @@
...
+if [[ -z "$arch" ]] && [[ "$(arch)" == "aarch64" ]]; then
+  arch="arm64"
+else
+  arch="x64_64"
```
Comment (orien) [inline hooks/post-checkout]:
```suggestion
  arch="x86_64"
```

**Review by orien — COMMENTED** [review]
I'm uneasy with these binaries included in the source. It's not clear where they originate from and an observer is left with questions:

- Where is the source repo?
- What version/release is running?
- What is the update process?
- Are they trustworthy?

To avoid this confusion, is it possible for the script to download the binaries from a reliable site at runtime?

**Comment by stevehodgkiss** [comment]
> I'm uneasy with these binaries included in the source.

Yeah, I agree. I don't think this is the right place for them either.

> is it possible for the script to download the binaries from a reliable site at runtime?

It feels to me like this software should be baked into the build agent AMI itself, and this plugin uses whatever version is available/installed on the agent, similar to how the docker plugins expect docker to be installed and available.

In terms of how the build agent AMI could install it, it could be either downloaded from a Github release (when available), or compiled from a source tag.

File: hooks/post-checkout (hunk 1 above)
Comment (stevehodgkiss) [reply hooks/post-checkout]: Yeah, that was unusable. Removed.

**Review by orien — APPROVED** [review]
I think this is good for now. When the upstream `ejson2env` project accepts responsibility for providing arm64 binaries, we can then circle back and make some improvements. (Downloading the release either via this plugin, or installed directly on the agent.

File: hooks/post-checkout [hunk 3]
```diff
@@ -3,7 +3,19 @@
 set -euo pipefail
 
 EJSON2ENV_PLUGIN_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")"; cd ..; pwd)
-EJSON2ENV_BIN_PATH="$EJSON2ENV_PLUGIN_ROOT/ejson2env"
+arch="$(arch)"
+case "$arch" in
+  arm*|aarch64)
+    arch="arm64";;
```
Comment (orien) [inline hooks/post-checkout:9]: Do we need a `break` in here? I suspect the flow will always fall through if we don't

```suggestion
    arch="arm64"; break;;
```

Comment (stevehodgkiss) [reply hooks/post-checkout:9]: `;;` breaks the case statement.  I've also confirmed it works on both architectures. Docs:

> Each clause must be terminated with ‘;;’, ‘;&’, or ‘;;&’.
> There may be an arbitrary number of case clauses, each terminated by a ‘;;’, ‘;&’, or ‘;;&’.

https://www.gnu.org/software/bash/manual/html_node/Conditional-Constructs.html

Comment (orien) [reply hooks/post-checkout:9]: No, I must be confusing this with something else.

It works fine in my experimentation.

Comment (orien) [reply hooks/post-checkout:9]: Thanks @stevehodgkiss 👍

**Review by orien — APPROVED** [review]
🚀

**Pull request merged**